src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -218,7 +167,131 @@ private async Task UpdateJobRecordAsync(CancellationToken cancellationToken)
+                            progress.TriggeringResourceId,
+                            null,
+                            queryParametersList,
+                            cancellationToken);
+                    }
+                    else
+                    {
+                        switch (_exportJobRecord.ExportType)
+                        {
+                            case ExportJobType.All:
+                                searchResult = await searchService.Value.SearchAsync(
+                                    null,
+                                    queryParametersList,
+                                    cancellationToken);
+                                break;
+                            case ExportJobType.Patient:
+                                searchResult = await searchService.Value.SearchAsync(
+                                    KnownResourceTypes.Patient,
+                                    queryParametersList,
+                                    cancellationToken);
+                                break;
+                        }
+                    }
+                }
+
+                if (_exportJobRecord.ExportType == ExportJobType.Patient && !compartmentSearch)
+                {
+                    uint resultIndex = 0;
+                    foreach (SearchResultEntry result in searchResult.Results)
+                    {
+                        // If a job is resumed in the middle of processing patient compartment resources it will skip patients it has already exported compartment information for.
+                        if (progress.SubSearch != null && result.Resource.ResourceId != progress.SubSearch.TriggeringResourceId)
+                        {
+                            continue;
+                        }
+
+                        if (progress.SubSearch == null)
+                        {
+                            progress.NewSubSearch(result.Resource.ResourceId);
+                        }
+
+                        await RunExportSearch(exportJobConfiguration, progress.SubSearch, sharedQueryParametersList, cancellationToken, pagesWaitingForCommit, currentBatchId + ":" + resultIndex.ToString("d6"), true);
+                        resultIndex++;
+
+                        progress.ClearSubSearch();
+                    }
+                }
+
+                await ProcessSearchResultsAsync(searchResult.Results, currentBatchId, cancellationToken);
+
+                if (searchResult.ContinuationToken == null)
+                {
+                    // No more continuation token, we are done.
+                    break;
+                }
+
+                // Update the continuation token in local cache and queryParams.
+                // We will add or udpate the continuation token to the end of the query parameters list.
+                progress.UpdateContinuationToken(searchResult.ContinuationToken);
+                pagesWaitingForCommit++;
```

Comment:
`pagesWaitingForCommit` doesn't seem to be used anywhere?

Reply:
Good catch. I was messing around with the pages to commit settings and this is a remnant.